### ISSUE TITLE
Indenting multiline text output

### ISSUE THREAD

**Issue by AnLogLabster** [scriban/scriban#398]
I have been struggling with getting neat, properly indented output when using Scriban in XML.

My simplified use case is the following:

```
{{~
# Define a variable with a multiline string as its value
li = "<li/>
<li/>
<li/>"
~}}

Variable printing statement is indented, only first line ends up indented
<ul>
  {{ li }}
</ul>
```

Its output is:
```
Variable printing statement is indented, only first line ends up indented
<ul>
  <li/>
<li/>
<li/>
</ul>
```

I have this problem with any text printing, this includes that from function calls and `include`. The desired output of the second block is:
```
<ul>
  <li/>
  <li/>
  <li/>
</ul>
```

This seriously impacts my ability to produce readable XML.

I know that in this simplified example I can get the right indenting by printing the li's in separate statements. But this will not help me in my real usecases. One of which is using `include` on a block of XML and expecting it to have the right indentation, based off of the first line. In other words, my need is **the ability to indent all of the lines in multiline string output**.

Currently, my last resort is exporting a NET method that would indent everything up to n spaces. But this would be quite bad from a developer ergonomics PoV.

I wasn't able to grok entirely how related my issue is to #253 . Apologies if I missed some obvious way of doing this.

**Comment by xoofx** [comment]
Indeed, that's the behavior and I can't change this - as that would be a breaking change. Some other users would want that it never indents and respects exactly what is in the string.

Usually when you are using Scriban for such case, you have something like that:

```
<ul>
{{~ for li in items ~}}
    {{ li }}
{{~ end ~}}
</ul>
```

Maybe a function could be introduced:

```
<ul>
    {{ li | indent_new_lines }}
</ul>
```

But this function does not exist. It would have to be implemented.

**Comment by AnLogLabster** [comment]
Thank you for the very swift answer!! I hadn't considered piping for some reason, that does have better readability than I was thinking.

Would a PR that adds a mode to `TemplateContext` that controls whether or not the output should be indented or not be at least in concept welcome? Or are there reasons why that approach would also not be good?

**Comment by xoofx** [comment]
> Would a PR that adds a mode to TemplateContext that controls whether or not the output should be indented or not be at least in concept welcome? Or are there reasons why that approach would also not be good?

Oh, that could be done as well at the `TemplateContext` level. PR welcome. Maybe be the functionality could moved from `include` to generalize it...

**Comment by xoofx** [comment]
I fixed it by commit 7eafa97, the [auto-indentation](https://github.com/scriban/scriban/blob/b6e558819d46146e052b297b1a2715201a32bc2e/doc/language.md#15-auto-indentation). It seems that it generalizes well, helped to remove the specificities for include and all the existing tests were passing. 
I made it on by default as I believe it is more the expected behavior for the default of a text templating, so It will likely impact some users but hopefully not too many. Let's see.

This should be part of `5.3.0+`.
